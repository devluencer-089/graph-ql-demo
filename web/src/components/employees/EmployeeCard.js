import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import Card, { CardActions, CardContent, CardHeader } from 'material-ui/Card';
import Button from 'material-ui/Button';
import Typography from 'material-ui/Typography';

const styles = theme => ({
  card: {
    minWidth: 275,
    margin: theme.spacing.unit
  }
});

function EmployeeCard(props) {
  const { classes, employee } = props;
  const fullname = `${employee.firstname} ${employee.lastname}`;
  const level = translateLevel(employee.level);
  return (
    <Card className={classes.card}>
      <CardHeader title={fullname} subheader={level} />
      <CardContent>
        <Typography component="p">{employee.phoneNumber}</Typography>
      </CardContent>
      <CardActions>
        <Button size="small">Details</Button>
      </CardActions>
    </Card>
  );
}

EmployeeCard.propTypes = {
  classes: PropTypes.object.isRequired,
  employee: PropTypes.object.isRequired
};

function translateLevel(level) {
  switch (level) {
    case 'JUNIOR_DEVELOPER':
      return 'Junior Developer';
    case 'SENIOR_DEVELOPER':
      return 'Senior Developer';
    case 'ARCHITECT':
      return 'Architect';
    case 'LEAD_DEVELOPER':
      return 'Lead Developer';
    default:
      return 'Unpositioned';
  }
}

export default withStyles(styles)(EmployeeCard);
