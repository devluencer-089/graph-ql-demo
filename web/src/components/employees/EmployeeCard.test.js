import React from 'react';
import { createShallow } from 'material-ui/test-utils';
import EmployeeCard from './EmployeeCard';

describe('EmployeeCard Component', () => {
  let shallow = createShallow({ dive: true });
  let employee = {
    id: '3',
    level: 'SENIOR_DEVELOPER',
    firstname: 'Si',
    lastname: 'Tran'
  };

  it('should match snapshot', () => {
    let eut = shallow(<EmployeeCard employee={employee} />);

    expect(eut).toMatchSnapshot();
  });
});
