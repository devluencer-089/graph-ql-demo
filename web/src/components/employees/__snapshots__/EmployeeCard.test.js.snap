// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeCard Component should match snapshot 1`] = `
<Card
  className="EmployeeCard-card-1"
  raised={false}
>
  <WithStyles(CardHeader)
    subheader="Senior Developer"
    title="Si Tran"
  />
  <WithStyles(CardContent)>
    <WithStyles(Typography)
      component="p"
    />
  </WithStyles(CardContent)>
  <WithStyles(CardActions)>
    <WithStyles(Button)
      size="small"
    >
      Details
    </WithStyles(Button)>
  </WithStyles(CardActions)>
</Card>
`;
